dipy/segment/tissue.py

```diff
@@ -125,3 +125,98 @@ def classify(self, image, nclasses, beta, *, tolerance=1e-05, max_iter=100):
         PVE = PVE[..., 1:]
 
         return initial_segmentation, final_segmentation, PVE
+
+
+def compute_directional_average(data, bvals, b0_thresh=10):
+    """
+    Compute the mean signal for each unique b-value shell and fit a linear model.
+
+    Parameters
+    ----------
+        data : ndarray
+            The diffusion MRI data.
+        bvals : ndarray
+            The b-values corresponding to the diffusion data.
+
+    Returns
+    -------
+        P : float
+            The slope of the linear model.
+        V : float
+            The intercept of the linear model.
+    """
+    unique_bvals = np.unique(bvals)
+    # Mean signal for b=0 (non-diffusion weighted)
+    s0 = data[bvals < b0_thresh].mean()
+
+    if len(unique_bvals) <= 2:
+        raise ValueError("Insufficient unique b-values for fitting.")
+
+    # If the mean signal for b=0 is too low, return 0, 0
+    if data[bvals < b0_thresh].mean() < 50:
```

Comment:
why 50 ? it should be documented in the docstring. Or maybe it can be an input parameter for the user.

Reply:
Added extra low signal threshold for users but kept it as 50 default as It is a tricky parameter to identify.